lib/package.gi

```diff
@@ -159,6 +159,172 @@ InstallGlobalFunction( SetPackageInfo, function( record )
     end );
 
 
+#############################################################################
+##
+#F  FindPackageInfosInSubdirectories( pkgdir, name )
+##
+##  Finds all PackageInfos in subdirectories of directory name in
+##  directory pkgdir, returns a list of their paths.
+##
+BindGlobal( "FindPackageInfosInSubdirectories", function( pkgdir, name )
+    local pkgpath, file, files, subdir;
+    
+    if name in [ ".", ".." ] then
+      return [];
+    fi;
+
+    pkgpath:= Filename( [ pkgdir ], name );
+
+    # This can be 'fail' if 'name' is a void link.
+    if pkgpath = fail then
+      return [];
+    fi;
+
+    if not IsDirectoryPath( pkgpath ) then
+      return [];
+    fi;
+
+    file:= Filename( [ pkgdir ],
+                      Concatenation( name, "/PackageInfo.g" ) );
+
+    if file <> fail then
+      return [ [ file, name ] ];
+    fi;
+
+    files := [];
+    # Perhaps some subdirectories contain `PackageInfo.g' files.
+    for subdir in Set( DirectoryContents( pkgpath ) ) do
+      if subdir in [ ".", ".." ] then
+        continue;
+      fi;
+      file:= Filename( [ pkgdir ],
+          Concatenation( name, "/", subdir, "/PackageInfo.g" ) );
+      if file <> fail then
+        Add( files,
+              [ file, Concatenation( name, "/", subdir ) ] );
+      fi;
+    od;
```

Comment:
This loop could now also be simplified, e.g. to something like this:
```gap
  files := Filtered(SubdirectoriesOfPath(pkgpath), dir -> fail <> Filename([Directory(dir)], "PackageInfo.g"));
```

Reply:
The new loop is a bit different now.